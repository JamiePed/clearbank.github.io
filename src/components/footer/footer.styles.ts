import styled from 'styled-components'
import { Box } from 'rebass'

import { colors, breakpoints } from 'src/components/theme'

export const Container: any = styled(Box)`
  display: flex;
  flex-direction: column;
  padding-top: 20px;
  border-top: dashed 1px ${colors.alto};

  @media screen and (min-width: ${breakpoints.large}) {
    align-items: start;
    flex-direction: row;
    justify-content: space-between;
  }

  @media screen and (min-width: ${breakpoints.large}) {
    padding: 30px;
  }
`

export const LogoWrapper: any = styled.div`
  max-width: 170px;
  display: block;
  margin: auto;
  margin-bottom: 30px;
  margin-top: 30px;

  @media screen and (min-width: ${breakpoints.large}) {
    margin-top: 0;
  }
`

export const CopywriteWrapper = styled.div`
  text-align: center;
  margin-bottom: 40px;

  @media screen and (min-width: ${breakpoints.medium}) {
    margin-top: 30px;
  }

  @media screen and (min-width: ${breakpoints.large}) {
    padding-right: 50px; // space for to top button
  }
`

export const CopyContainer = styled.div`
  padding: 0 20px;
`

export const Copy = styled.p`
  display: block;
  width: 100%;
  color: ${colors.brandSecondaryDarker};
  font-size: 14px;

  @media screen and (min-width: ${breakpoints.large}) {
    padding-right: 55px;

    &:first-child {
      margin-top: 30px;
    }
  }
`

export const Copywrite = styled.span`
  white-space: nowrap;
  color: ${colors.brandSecondaryDarker};
`
