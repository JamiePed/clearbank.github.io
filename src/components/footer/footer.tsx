import React from 'react'

import Logo from 'src/components/logo'

import * as Styles from './footer.styles'

const Footer: React.FC<{}> = ({ items }) => (
  <Styles.Container role='contentinfo'>
    <Styles.MenuContainer>
      {items.map(item => (
        <Styles.ColumnContainer key={item.menuItem.title}>
          <Styles.Title>
            {item.menuItem.title}
          </Styles.Title>
          <Styles.List>
            {item.subMenuItems.map(subMenuItem => (
              <Styles.ListItem key={subMenuItem.node.fields.slug}>
                <Styles.SubTitle>
                  {subMenuItem.node.fields.title}
                </Styles.SubTitle>
              </Styles.ListItem>  
            ))}
          </Styles.List>
        </Styles.ColumnContainer>
      ))}
    </Styles.MenuContainer> 
    <Styles.CopywriteWrapper>
      <Styles.LogoWrapper>
        <Logo inverted />
      </Styles.LogoWrapper>

      <Styles.Copywrite className='long-primer'>
        Copyright ClearBank Ltd. Authorised by the Prudential Regulation Authority and regulated by the Financial Conduct Authority and the Prudential Regulation Authority
        (Financial Services Register number: 754568). Registered Address: ClearBank, 4th Floor, Prologue Works, 25 Marsh St, Bristol BS1 4AX
      </Styles.Copywrite>
    </Styles.CopywriteWrapper>
  </Styles.Container>
)

export default Footer
