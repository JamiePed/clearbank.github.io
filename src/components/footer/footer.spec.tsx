import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './footer'

import * as Types from './footer.types'

let component: ReturnType<typeof render> = null
let fragment: DocumentFragment = null
let root: HTMLElement = null

const defaultProps: Types.FooterProps = {}

beforeEach(() => {
  component = render(<Component {...defaultProps} children='children' />)

  fragment = component.asFragment()
  root = component.container.firstChild
})

afterEach(() => {
  fragment = null
  root = null
})

describe('Component', () => {
  test('matches snapshot', () => {
    expect(root).toMatchSnapshot()
  })

  test('has child elements', () => {
    expect(root.hasChildNodes()).toBeTruthy()
  })

  test('has offical company name in text', () => {
    expect(root.textContent.includes('Copyright ClearBank Ltd.')).toBeTruthy()
  })

  test('has address in text', () => {
    expect(
      root.textContent.includes('ClearBank, 4th Floor, Prologue Works, 25 Marsh St, Bristol BS1 4AX.')
    ).toBeTruthy()
  })

  test('has current year in text', () => {
    const currentYear = new Date().getFullYear().toString()

    expect(root.textContent.includes(currentYear)).toBeTruthy()
  })

  test('is not empty', () => {
    expect(fragment.textContent.length).toBeTruthy()
  })
})
