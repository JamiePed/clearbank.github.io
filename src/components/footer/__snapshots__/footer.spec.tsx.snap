// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
.c3 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0 7%;
  padding: 0 5%;
  border-bottom: 1px solid #D4D2E3;
}

.c2 {
  max-width: 170px;
  display: block;
  margin: 40px 0px 20px 40px;
}

.c1 {
  text-align: center;
  margin-bottom: 40px;
}

.c4 {
  max-width: 70%;
  margin-left: auto;
  text-align: right;
  color: #6C7F98;
}

<div
  class="css-1y75s01"
  role="contentinfo"
>
  <div
    class="c0"
  />
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <img
          alt="ClearBank"
          src=""
        />
      </div>
    </div>
    <p
      class="c4 long-primer"
    >
      Copyright ClearBank Ltd. Authorised by the Prudential Regulation Authority and regulated by the Financial Conduct Authority and the Prudential Regulation Authority (Financial Services Register number: 754568). Registered Address: ClearBank, 4th Floor, Prologue Works, 25 Marsh St, Bristol BS1 4AX
    </p>
  </div>
</div>
`;
